packages/web-react/src/components/Modal/demo/ModalScrollingLongContent.tsx

```diff
@@ -216,104 +213,108 @@ const ModalScrollingLongContent = () => {
-                      UNSAFE_style={{
-                        ['--grid-item-column-start' as string]: 1,
-                        ['--grid-item-column-end' as string]: 6,
-                        ['--grid-item-column-end-tablet' as string]: 4,
-                      }}
-                    >
-                      Max height
-                    </Label>
-                    <GridItem
-                      columnStart={{ mobile: 6, tablet: 4 }}
-                      columnEnd={{ mobile: 13, tablet: 7 }}
-                      elementType="span"
-                      id={`custom-max-height-${breakpoint}-value`}
-                    >
-                      {isCustomHeightEnabled[breakpoint] ? `${heightValue.maxHeight[breakpoint]} px` : '—'}
-                    </GridItem>
-                    <input
-                      autoComplete="off"
-                      className="GridItem"
-                      defaultValue={heightValue.maxHeight[breakpoint]}
-                      disabled={!isCustomHeightEnabled[breakpoint]}
-                      id={`custom-max-height-${breakpoint}`}
-                      max="1000"
-                      min="200"
-                      onChange={(event) => handleMaxHeightChange(event, breakpoint)}
-                      step="100"
-                      style={{
-                        ['--grid-item-column-end' as string]: 13,
-                        ['--grid-item-column-start' as string]: 1,
-                        ['--grid-item-column-start-tablet' as string]: 7,
-                      }}
-                      type="range"
+              {Object.entries(breakpointControls).map(([breakpointKey, { marginBottom, className }]) => {
+                const breakpoint = breakpointKey as BreakpointToken;
+
+                return (
+                  <Stack
+                    elementType="fieldset"
+                    hasSpacing
+                    key={breakpoint}
+                    marginBottom={marginBottom as Partial<Record<string, SpaceToken>>}
+                    UNSAFE_className={className}
+                    UNSAFE_style={{ border: 0 }}
+                  >
+                    <legend hidden>{breakpoint.charAt(0).toUpperCase() + breakpoint.slice(1)}</legend>
+                    <Checkbox
+                      id={`custom-height-${breakpoint}-enabled`}
+                      isChecked={isCustomHeightEnabled[breakpoint]}
+                      label={breakpoint.charAt(0).toUpperCase() + breakpoint.slice(1)}
+                      onChange={(event) =>
+                        setIsCustomHeightEnabled((prevState) => ({
+                          ...prevState,
+                          [breakpoint]: (event.target as HTMLInputElement).checked,
+                        }))
+                      }
                     />
-                  </Grid>
-                </Stack>
-              ))}
+                    <Grid UNSAFE_style={{ columnGap: 'var(--spirit-space-600)' }}>
```

Comment:
I think we have props for this now. But I will leave it up to you, because it is definitely outside of the scope of this issue :)